subprojects/dependency-management/src/integTest/groovy/org/gradle/api/artifacts/transform/ArtifactTransformBuildScanIntegrationTest.groovy

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.artifacts.transform
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.internal.scan.config.fixtures.BuildScanPluginFixture
+
+class ArtifactTransformBuildScanIntegrationTest extends AbstractIntegrationSpec {
+    def fixture = new BuildScanPluginFixture(testDirectory, mavenRepo, createExecuter())
+
+    def setup() {
+        settingsFile << fixture.pluginManagement() << """
+            include 'lib'
+            include 'util'
+        """
+        fixture.logConfig = true
+        fixture.publishDummyBuildScanPluginNow()
+    }
+
+    def "transform works with build scan"() {
+        given:
+        buildFile << """
```

Comment:
Can we now re-use code from the other artifact transform tasks so we don't need to add this large code block again?